types/src/on_chain_config/consensus_config.rs

```diff
@@ -12,6 +12,7 @@ use serde::{Deserialize, Serialize};
 #[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
 pub enum OnChainConsensusConfig {
     V1(ConsensusConfigV1),
+    V2(ConsensusConfigV2),
```

Comment:
noob question: why do we have to create a new config version in this case? Can't we add the new flag to V1 and annotate it with `#serde(default = ...)` to maintain backwards compatibility? in other words, is there a need for V2 config?

Reply:
+1. cc @igor-aptos

Reply:
Awesome. So if I make the change and see compat test succeed we should be good, right?

Reply:
Discussed with @zekun000 offline. Since this is using BCS, it will fail because BCS is not extendable (to be canonical).

The on chain config probably doesn't need to use BCS, but it's using BCS today.

Reply:
I think it makes sense to move away from bcs encoding to json or something more flexible. the data stored on-chain just as vector, and the interpretation is off-chain so we can be more flexible here. but need to do a migration first

Reply:
Oh that's interesting. So the enabling will be like below?
```
    pub fn enable_quorum_store(&self) -> bool {
        match &self {
            OnChainConsensusConfig::V1(_) => false,
            OnChainConsensusConfig::V2(_) => true,
        }
    }
```

I like that it avoids a lot of repeating. If in emergency we need to revert, we just push a new V1 config, right?

Reply:
yes